src/main/cpp/src/json_parser.hpp

```diff
@@ -524,23 +536,26 @@ class json_parser {
     char* copy_destination,
     write_style w_style)
   {
-    return try_parse_quoted_string(
-      str_pos,
-      '\'',
-      to_match_str_pos,  // match str pos, nullptr means do not match
-      to_match_str_end,  // match str end
-      copy_destination,  // copy destination while parsing, nullptr means do not copy
-      w_style);
+    return try_parse_quoted_string(str_pos,
+                                   '\'',
```

Comment:
I'm not sure where this function will run (host or device?), as a lot of functions here are marked as `CUDF_HOST_DEVICE`. Generally we should not do that, only `__device__` or without marking (host functions).

Anyway, this may crash, since it is passing a `char*` pointer that may point to memory and that pointer may be passed into device function. If this is a host function then it's fine. If it is a device function, pass a `string_scalar` instead.

Reply:
> I'm not sure where this function will run (host or device?)

Refer to this [comment](https://github.com/NVIDIA/spark-rapids-jni/pull/1868#discussion_r1530015452)